lib/src/command/add.dart

```diff
@@ -259,12 +336,12 @@ class AddCommand extends PubCommand {
     );
   }
 
-  /// Parse [package] to return the corresponding [PackageRange], as well as its
-  /// representation in `pubspec.yaml`.
+  /// Parse [package] to return the corresponding [_ParseResult] using the
+  /// arguments given in [argResults] to configure the description.
   ///
   /// [package] must be written in the format
-  /// `<package-name>[:<version-constraint>]`, where quotations should be used
-  /// if necessary.
+  /// `[dev:]<package-name>[:<version-constraint>]`, where quotations should be
```

Comment:
```suggestion
  /// `<package-name>[:<version-constraint>]`, where quotations should be
```

You've already parse `dev:` when you get here.